hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -65,6 +68,14 @@ private synchronized void init() {
               .removalListener((key, value, cause) -> ((HiveClientPool) value).close())
               .build();
     }
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      @Override
+      public void run() {
+        LOG.info("Before clientPool closed, clientPool().getCurrentSize() = {}", clientPool().getCurrentSize());
```

Comment:
Nit: Would this be better as a lower log level? Additionally, would it make more sense to format this as human readable / formatted text as opposed to references to functions (which might be awkward for end users)?

For example: `The CachedClientPool has a size of {}`.